### PR TITLE
gh-782: Add benchmarks for `glass.fields`/`glass.harmonics`

### PR DESCRIPTION
**Pull request by connoraird**
# Description

Add benchmark tests for `glass.fields` and `glass.harmonics` for any functions which have both been identified as worth benchmarking in #790 and do not return Generators. Generator functions will be handled in #809.

Closes: #782, #813

## Checks

- [x] Is your code passing linting?
- [x] Is your code passing tests?
- [x] Have you added additional tests (if required)?
- [ ] Have you modified/extended the documentation (if required)?
- [ ] Have you added a one-liner changelog entry above (if required)?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -225,6 +225,8 @@ lint.per-file-ignores = {"__init__.py" = [
     "INP001", # implicit-namespace-package
     "PLR2004", # magic-value-comparison
     "S101", # assert
+    "S311", # Standard pseudo-random generators are not suitable for cryptographic purposes
+    "SLF001", # Private member accessed (in tests we want to access private members)
```
Comment (paddyroddy) [inline pyproject.toml]: Just use slug like elsewhere

```suggestion
    "S311", # suspicious-non-cryptographic-random-usage
    "SLF001", # private-member-access
```

File: tests/benchmarks/test_fields.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """Benchmarks for glass.fields.getcl with lmax of 0."""
+    range_size = benchmark_scale_factor
+    # make a mock Cls array with the index pairs as entries
+    cls = [
+        xp.asarray([i, j], dtype=xp.float64)
+        for i in range(range_size)
+        for j in range(i, -1, -1)
+    ]
+
+    random_i = randrange(range_size)
```
[the remark this answers is not in the document]
Comment (connoraird) [reply tests/benchmarks/test_fields.py]: The tests in core run through a load of indices and check they are all accessed correctly but we can only call the function once if we want to benchmark so this randomly selects indices instead.

File: tests/benchmarks/test_harmonics.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    xp: ModuleType,
+    benchmark: BenchmarkFixture,
+    benchmark_scale_factor: int,
+) -> None:
+    """Benchmarks for glass.harmonics.multalm."""
+    # Call jax version of iternorm once jax version is written
+    if xp.__name__ == "jax.numpy":
+        pytest.skip("Arrays in multalm are not immutable, so do not support jax")
+
+    scale_factor = benchmark_scale_factor * 100
```
[the remark this answers is not in the document]
Comment (connoraird) [reply tests/benchmarks/test_harmonics.py]: I decided to remove it as we had a different value already in main.

**Comment by paddyroddy** [comment]
🙏 remember labels

File: tests/benchmarks/test_fields.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+from random import randrange
```
Comment (paddyroddy) [inline tests/benchmarks/test_fields.py]: To be consistent with eleswhere can we `import random` then use `random.randrange`? The benefit is at a glance you can see where it comes from.

File: tests/benchmarks/test_fields.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    result = benchmark(
+        glass.fields.getcl,
+        cls,
+        random_i,
+        random_j,
+        lmax=lmax,
+    )
+    expected = xp.zeros((lmax - 1,), dtype=xp.float64)
+    assert result.size == lmax + 1
+    np.testing.assert_allclose(result[2:], expected)
```
Comment (paddyroddy) [inline tests/benchmarks/test_fields.py]: `compare` fixture
